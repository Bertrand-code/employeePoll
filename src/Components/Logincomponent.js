import React from "react";
import { connect } from "react-redux";
import { Navigate } from "react-router-dom";
import { userLogin } from "../Store/authReducer";

function Logincomponent({ login, user, auth }) {
  const [username, setUsername] = React.useState("sarahedo");
  const [password, setPassword] = React.useState("password123");

  const handleSubmit = (e) => {
    e.preventDefault();
    login({ username, password });
  };
  if (auth) return <Navigate to={"/"} replace />;

  return (
    <div>
      <form onSubmit={handleSubmit} data-testid="loginForm">
        <label>
          Username:
          <input
            type="text"
            value={username}
            onChange={(e) => setUsername(e.target.value)}
          />
          <br />
        </label>
        <br />
        <label>
          Password:
          <input
            type="password"
            value={password}
            onChange={(e) => setPassword(e.target.value)}
          />
        </label>
        <br />
        <button> Login </button>
      </form>
      {JSON.stringify(user)}
    </div>
  );
}

const mapDispatchToProps = {
  login: userLogin,
};
const mapStateToProps = (state) => ({
  auth: state.authStore.auth,
});

export default connect(mapStateToProps, mapDispatchToProps)(Logincomponent);
