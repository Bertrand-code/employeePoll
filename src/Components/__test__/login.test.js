import { render, cleanup, screen } from "@testing-library/react";
import Logincomponent from "../Logincomponent";
import { BrowserRouter, Route, Routes } from "react-router-dom";
import { Provider } from "react-redux";
import { store } from "../../Store";

afterEach(() => {
  cleanup();
});

const TestLogin = () => {
  return (
    <BrowserRouter>
      <Provider store={store}>
        <Routes>
          <Route path="/" element={<Logincomponent />} />
        </Routes>
      </Provider>
    </BrowserRouter>
  );
};

describe("Logincomponent", () => {
  it("should render and match screenshot", () => {
    const { container } = render(<TestLogin />);
    expect(container).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it("should render form", () => {
    render(<TestLogin />);
    expect(screen.getByTestId("loginForm")).toBeInTheDocument();
    expect(
      screen.getByRole("textbox", {
        type: "password",
      })
    ).toBeInTheDocument();
    expect(screen.getByText("Login")).toBeInTheDocument();
  });
});
