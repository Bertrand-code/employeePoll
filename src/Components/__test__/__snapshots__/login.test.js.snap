// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logincomponent should render and match screenshot 1`] = `
<div>
  <div>
    <form
      data-testid="loginForm"
    >
      <label>
        Username:
        <input
          type="text"
          value="sarahedo"
        />
        <br />
      </label>
      <br />
      <label>
        Password:
        <input
          type="password"
          value="password123"
        />
      </label>
      <br />
      <button>
         Login 
      </button>
    </form>
  </div>
</div>
`;
